src/pulse3D/metrics.py

```diff
@@ -154,22 +154,27 @@ def calculate_amplitudes(
         Returns:
             Pandas Series of float values representing the amplitude of each twitch
         """
+        twitch_indices_hashable = HashableDataFrame(DataFrame.from_dict(twitch_indices).T)
```

Comment:
We need this to be fast enough to run in live view but unfortunately initializing dataframes is quite slow, so we need to stick to just using regular dicts here

Reply:
`TwitchWidth.calculate_twitch_widths` requires a HashableDataframe to be passed as the `twitch_indices`. I can move it up to its parent function?

Reply:
I say let's leave it for now and assess the performance in the MA Controller later